js/renderer/texturesprite_renderer.js

```diff
@@ -180,13 +252,88 @@ LayerScope.SpriteView = {
         this._colors.push($cdiv);
       }
 
+      let color = o;
       $sprite.on("click", function() {
         LayerScope.MessageCenter.fire("buffer.select", layerID.toString());
+        LayerScope.MessageCenter.fire("color.select", color);
       });
 
       $sprite.append($bgdiv);
-      $panel.append($sprite);
+      // Last step, push this new sprite to sorted array.
+      this._sortSpriteByLayerID($sprite);
     }
   }
 };
 
+// Regist SpriteView into RendererNode
+LayerScope.RendererNode.register(LayerScope.SpriteView);
```

Comment:
You don't need to register LayerScope.SpriteView. bufferview_renderer.js did that for you already